### PR TITLE
WIP! - custom OSM server config for docker-compose

### PR DESCRIPTION
**Pull request by majdal**
**Note** This is work in progress - just opened the PR to make sure that I'm starting on the right foot. I'm not sure if this is good practice or not, so feel free to close and I'll re-open when I'm further along. 

Meant to address #2865 

A question though: Shall I also change the names in the user messages? e.g. the welcome message at: `backend/services/messaging/templates/welcome_message_en.txt`

### CODE REVIEW

**Comment by JorgeMartinezG** [comment]
I like this

**Comment by majdal** [comment]
Should also be checked against #2816 (docker improvements branch)

**Comment by geohacker** [comment]
Hi @majdal @willemarcel! Thanks so much @majdal for your work on this so far. This feature will be very useful for us as I'm planning to add TM4 into [osm-seed](https://github.com/developmentseed/osm-seed/). Primarily to be used within openhistoricalmap.org.

@willemarcel Could be perhaps put this for release soon? 
@majdal looks like you have completed most of what's needed. Do you need me to look at anything specific?

**Comment by majdal** [comment]
@geohacker there might be some more work to do after #2816 was merged, I can check next week maybe!

File: frontend/src/components/header/messages.js [hunk 1]
```diff
@@ -38,16 +39,16 @@ export default defineMessages({
   },
   AuthorizeTitle: {
     id: 'signUp.modal.authorize',
-    defaultMessage: 'Have you finished the registration on OpenStreetMap?',
+    defaultMessage: `Have you finished the registration on ${TM_ORG_NAME}?`,
```
Comment (willemarcel) [inline frontend/src/components/header/messages.js:42]: Inserting env vars on react-intl messages will generate a mess, and TM_ORG_NAME here is the organization that is deploying the TM instance (on HOT case it is "Humanitarian OpenStreetMap Team"). In my opinion, we should continue using OpenStreetMap. When you are deploying TM in a context of another OSM server, you will have the OpenStreetMap website with the same name, wouldn't you?

File: frontend/src/utils/editorsList.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
+import { ID_EDITOR_URL, POTLATCH2_EDITOR_URL } from '../config';
+
 export function getEditors(filterList, customEditor) {
   let editors = [
     {
       label: 'iD Editor',
       value: 'ID',
-      url: 'http://www.openstreetmap.org/edit?editor=id&',
+      url: `${OSM_SERVER_URL}&`,
```
Comment (willemarcel) [inline frontend/src/utils/editorsList.js:8]: OSM_SERVER_URL, ID_EDITOR_URL and POTLATCH2_EDITOR_URL are not being exported from `../config`

**Review by willemarcel — CHANGES_REQUESTED** [review]
@majdal let us know if you need help to fix the issues

**Comment by geohacker** [comment]
@willemarcel I pulled @majdal's work and addressed your comments in https://github.com/hotosm/tasking-manager/pull/3564 — hope that's alright!

**Pull request closed without merging**